tachyon/math/finite_fields/mersenne31/packed_mersenne31_avx512.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2022 The Plonky3 Authors
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.plonky3 and the LICENCE-APACHE.plonky3
+// file.
+
+#ifndef TACHYON_MATH_FINITE_FIELDS_MERSENNE31_PACKED_MERSENNE31_AVX512_H_
+#define TACHYON_MATH_FINITE_FIELDS_MERSENNE31_PACKED_MERSENNE31_AVX512_H_
+
+#include <stddef.h>
+
+#include "tachyon/math/finite_fields/mersenne31/mersenne31.h"
+#include "tachyon/math/finite_fields/packed_prime_field_base.h"
+
+namespace tachyon::math {
+
+class PackedMersenne31AVX512;
+
+template <>
+struct PackedPrimeFieldTraits<PackedMersenne31AVX512> {
+  using PrimeField = Mersenne31;
+
+  constexpr static size_t N = 16;
+};
+
+class PackedMersenne31AVX512
+    : public PackedPrimeFieldBase<PackedMersenne31AVX512> {
+ public:
+  using PrimeField = Mersenne31;
+
+  constexpr static size_t N = 16;
```

Comment:
How about setting `N` from PackedPrimeFieldTraits.
It seems that `N` has been defined multiple times